lib/grandpa/message_handler.go

```diff
@@ -49,15 +48,10 @@ func (h *MessageHandler) handleMessage(from peer.ID, m GrandpaMessage) (network.
 
 	switch msg := m.(type) {
 	case *VoteMessage:
-		// send vote message to grandpa service
-		h.grandpa.in <- &networkVoteMessage{
-			from: from,
-			msg:  msg,
-		}
-
+		h.grandpa.handleVoteMessage(from, msg)
 		return nil, nil
 	case *CommitMessage:
-		return nil, h.handleCommitMessage(msg)
+		return nil, h.grandpa.handleCommitMessage(msg)
```

Comment:
nit a bit out of scope, but let's wrap the error since we are touching this here

```suggestion
		err = h.grandpa.handleCommitMessage(msg)
		if err != nil {
			return nil, fmt.Errorf("handling commit message: %w", err)
		}
		return nil, nil
```